src/altinn_model_publisher/security/auth.py

```diff
@@ -0,0 +1,26 @@
+"""Security module with authorization middleware."""
+import logging
+import os
+from typing import Awaitable, Callable
+
+from aiohttp import web
+
+
+API_KEY = os.getenv("API_KEY", "secret-key")
+
+
+@web.middleware
+async def auth_middleware(
+    request: web.Request,
+    handler: Callable[[web.Request], Awaitable[web.StreamResponse]],
+) -> web.StreamResponse:
+    """Middleware that raises HTTPForbidden for unauthorized POST requests."""
+    try:
+        if "POST" in request.method:
```

Comment:
Dette kravet er litt mistenkelig, det er ikke bare POST som kan være beskyttet.

Reply:
Nei, men det er bare POST som skal være beskyttet på dette apiet

Reply:
Som middleware kjøres dette på alle requests, alt av POST får en ekstra sjekk om det har korrekt api-key, alt annet slipper gjennom som normalt. Hva er det du har problem med?

Reply:
Jeg forstår det, og det er det jeg mener ikke er helt riktig. Det beste er å annotere handler du ønsker å beskytte.

Reply:
Alternativt, legge til auth_middleware kun på nødvendig handler

Reply:
Etter som jeg fant så var det kun mulig med middleware i aiohttp som ble kjørt for absolutt alle requests. Men kan ikke skryte av å ha lest så veldig nøye i dokumentasjonen, jeg ser på det etter jul :+1:

Reply:
Etter som jeg finner så er det ikke mulig å legge til middleware som ikke kjøres for alle requests. Hvordan mente du vi skulle bruke annoteringer?

Ok med noe i denne duren? 1c3cdf6

Reply:
Tenkte på noe sånt, https://github.com/aio-libs/aiohttp/issues/3590#issuecomment-465493071 i utgangspunktet, men innser at det ikke er en del av aiohttp security. Din løsning funker også 👍